src/Confluent.Kafka/Consumer.cs

```diff
@@ -85,8 +85,13 @@ public class Consumer<TKey, TValue> : IDisposable
 
         public bool Consume(out Message<TKey, TValue> message, int millisecondsTimeout)
         {
+            if (backgroundPollThread.IsStarted)
+            {
+                throw new Exception("Poll/Consume cannot be called on a Consumer when a background poll thread running.");
```

Comment:
Maybe a more specific exception - like `InvalidOperationException` - would be a better fit here?